Add ver_log_modified_tables function

strk opened the pull request:
This is useful if table_version.tables_changed gets lost
for any reason.

I actually wonder if we could turn that table into a view, to
avoid the maintainance effort. \cc @palmerj

strk commented:
I want to improve this function to become idempotent (exclude records already present)

palmerj commented on sql/18-log_modified_tables.sql:
```diff
@@ -0,0 +1,34 @@
+BEGIN
+
+    FOR v_rec IN SELECT id, schema_name, table_name
+                 FROM table_version.versioned_tables
```
Maybe better to use the function to get the tables and account for `versioned = TRUE`

strk replied:
The `ver_get_versioned_tables` function does NOT return the table identifier, which we need for our case, so we cannot use it. I've pushed a commit adding the `versioned = TRUE` filter.

strk commented:
> Maybe better to use the function to get the tables and account for `versioned = TRUE`

Using a function would prevent the optimizer from doing a single
scan to each revision table.

palmerj commented:
This looks good to me. Please ensure the CI is up to date.

strk commented:
Thanks @palmerj, I've squashed and rebased the code against current master, added a CHANGELOG and noticed documentation is still missing. Better add documentation before merging this.
